### PR TITLE
Fix bug where compile/minify excluded were ignored.

### PR DESCRIPTION
**Pull request by aomarks**
My earlier refactor in https://github.com/Polymer/polymer-build/pull/292 was totally wrong with respect to filtering using the `compile.exclude` and `compile.minify` options.

Also do compile/minification in the right order (Babel plugins run first to last, but presets run last to first :/).

 - [x] CHANGELOG.md has been updated

### CODE REVIEW

**Comment by justinfagnani** [comment]
We should add a regression test

File: src/optimize-streams.ts [hunk 1]
```diff
@@ -113,28 +114,37 @@ export class GenericOptimizeTransform extends Transform {
...
         presets.push(babelPresetES2015NoModules);
         plugins.push(
-            externalHelpersPlugin,
-            objectRestSpreadTransform,
+            babelPluginExternalHelpers,
+            babelTransformPluginObjectRestSpread,
         );
       }
-      if (options.moduleResolution === 'node') {
+      if (nodeResolution) {
```
Comment (justinfagnani) [inline src/optimize-streams.ts]: why not  `if (options.moduleResolution === 'node') {`, or ` if (options.moduleResolution !== 'node') {`? At least do an explicit check against undefined.

**Comment by aomarks** [comment]
> We should add a regression test

Done. It is also covered by an integration test in polymer cli, but better to have one here.

File: src/optimize-streams.ts (hunk 1 above)
Comment (aomarks) [reply src/optimize-streams.ts]: Done.

**Pull request merged**